PDR-407 Sites PUT

cameronpettit opened the pull request:
Relates to #407 

Adds site PUT endpoint to API. Behaves analogously to the Protected Area Names PUT endpoint.

Employs new design philosophy of keeping the lambda handlers very lean, and moving logic based code to a layer.

marklise commented on pdr-api/layers/dataUtils/siteUtils.js:
```diff
@@ -1,5 +1,34 @@
+
+const ESTABLISHED_STATE = 'established';
+const HISTORICAL_STATE = 'historical';
+const REPEALED_STATE = 'repealed';
```
This is already exists as an exported item in dyanmodb.js.  We'll want to remove one of them.

marklise commented on pdr-api/layers/dataUtils/siteUtils.js:
```diff
@@ -27,6 +56,280 @@ async function getSitesForProtectedArea(identifier) {
+    const paSiteUpdate = {
+      TableName: TABLE_NAME,
+      Key: {
+        pk: { S: identifier.split('::')[0] },
+        sk: { S: `Site::${identifier.split('::')[2]}` }
+      },
```
Add some error handling in case the size of the split is not as expected before dereferencing the split like this

marklise commented on pdr-api/__tests__/mock_data.js:
```diff
@@ -35,6 +35,17 @@ exports.MockData = class {
+    legalName: 'Test Park 3',
+    displayName: 'Test Park 3',
+    phoneticName: 'tɛst pɑːk thriː',
+    status: 'repealed',
```
These and all other refs to usable constants should be imported.

marklise commented on pdr-api/handlers/sites/_identifier/PUT/index.js:
```diff
@@ -0,0 +1,35 @@
+    const currentTimeISO = getNowISO();
+
+    // Perform update.
+    const transactions = await createSitePutTransaction(identifier, body, queryParams?.updateType, user,
+      currentTimeISO);
```
This can stay as one line.

marklise commented on pdr-api/layers/dataUtils/siteUtils.js:
```diff
@@ -27,6 +56,280 @@ async function getSitesForProtectedArea(identifier) {
+      TableName: TABLE_NAME,
+      Key: {
+        pk: { S: identifier },
+        sk: { S: 'Details' }
```
We should `const` this as an exportable STRING

marklise reviewed (approved):
/w nits

cameronpettit commented:
Fixed nits in [f8ebc67](https://github.com/bcgov/parks-data-register/pull/435/commits/f8ebc67083598e637a865ebec8043e21f828595e). Added a new constants file `data-constants.js` in `dataUtils` layer to house project specific constants.
